bullet_train-fields/app/javascript/controllers/fields/file_field_controller.js

```diff
@@ -77,16 +82,61 @@ export default class extends Controller {
     this.removeFileFlagTarget.value = true;
   }
 
-  handleFileSelected() {
-    const statusText = this.selectFileButtonTarget.querySelector("span");
-    const icon = this.selectFileButtonTarget.querySelector("i");
-
+  handleFileSelected(event) {
     if (this.hasDownloadFileButtonTarget) {
       this.downloadFileButtonTarget.remove();
     }
 
-    statusText.innerText = "Select Another File";
-    icon.classList.remove("ti-upload");
-    icon.classList.add("ti-check");
+    const files = event.target.files;
+    this.updateSelectedFileList(files);
+  }
+
+  cancelFileUpload(event){
+    const fileInput = this.fileFieldTarget;
+    const files = fileInput.files;
+    const fileToCancel = event.target.dataset.filename;
+
+    const dt = new DataTransfer();
+    for (const file of files) {
+      if(file.name != fileToCancel){
+        dt.items.add(file);
+      }
+    }
+    fileInput.files = dt.files;
+    this.updateSelectedFileList(dt.files);
   }
+
+  updateSelectedFileList(files){
+    const statusText = this.selectFileButtonTarget.querySelector("span");
+    const icon = this.selectFileButtonTarget.querySelector("i");
+
+    this.selectedFileListTarget.innerHTML = "";
+
+    if(files.length){
+      for (const file of files) {
+        let fileElement = document.createElement('div');
+        fileElement.classList.add('py-1', 'flex', 'flex-wrap', 'items-center');
```

Comment:
@pascallaliberte I think this is a great suggestion, but I'd like to treat it as a separate refactoring issue. Currently we manually manipulate various `classList` attributes all over this controller, and I was just following the existing pattern. It would definitely go good to do what you've suggested but I think it might be messy to try to fix all of the other instances as part of this PR. I'll open an issue for the refactoring.

Reply:
Actually, I ended up going the `template` route, which kind of moots this issue, at least in this particular spot. Would probably still be good to refactor some other spots to use Stimulus style class handling.